src/store/modules/contacts.js

```diff
@@ -132,45 +129,26 @@ export default {
     },
     async refresh ({ commit, rootGetters, getters }, addr) {
       // Make this generic over networks
+      const oldContactInfo = getters['getContact'](addr)
+      const updateInterval = getters['getUpdateInterval']
+      const now = moment()
+      if (oldContactInfo && moment(oldContactInfo.lastUpdate).add(updateInterval, 'milliseconds').isAfter(now)) {
```

Comment:
Upon receiving a message we first insert a dummy/default contact while the contact is being retrieved, immediately after this `refresh` method is then called.

This dummy contact is given here:
https://github.com/cashweb/stamp/blob/5590ffcfad8f8065d0ee12fc27421648b85d9adc/src/utils/constants.js#L24-L33

I'm unsure of what `moment(oldContactInfo.lastUpdate)` yields when `oldContactInfo.lastUpdate` is undefined but it's probably wise to set `lastUpdate` in this dummy contact to an initial value in order to ensure this condition is not triggered and the refresh is halted.